pkg/k8s/synced/resources.go

```diff
@@ -21,6 +24,25 @@ type Resources struct {
 	resources map[string]<-chan struct{}
 	// stopWait contains the result of cache.WaitForCacheSync
 	stopWait map[string]bool
+
+	// timeSinceLastEvent contains the time each resource last received an event.
+	timeSinceLastEvent map[string]time.Time
+}
+
+func (r *Resources) getTimeOfLastEvent(resource string) (when time.Time, never bool) {
+	r.RLock()
+	defer r.RUnlock()
+	t, ok := r.timeSinceLastEvent[resource]
+	if !ok {
+		return time.Time{}, true
+	}
+	return t, false
+}
+
+func (r *Resources) Event(resource string) {
+	r.Lock()
+	defer r.Unlock()
+	r.timeSinceLastEvent[resource] = time.Now()
```

Comment:
Yeah, good question, that makes sense, ill move the time up before grabbing the lock.

In general, my expectations for time accuracy where pretty low, since the underlying client-go library function that checks for a cache to be synced actually just polls every 100ms: https://github.com/kubernetes/client-go/blob/v0.24.1/tools/cache/shared_informer.go#L270,  as well as the context here being a startup timeout that is generally going to be in the order of minutes.